store.go

```diff
@@ -218,11 +228,18 @@ func (s *memoryStore[K, T]) Edge(sourceHash, targetHash K) (Edge[K], error) {
 	return edge, nil
 }
 
+func (s *memoryStore[K, T]) EdgeCount() (int, error) {
+	s.lock.RLock()
+	defer s.lock.RUnlock()
+
+	return s.edgeCount, nil
```

Comment:
Is there a specific reason why you chose not to use something like `len(s.outEdges)` here?

Reply:
`len(s.outEdges)` will only return the correct results some of the time.  For example

```go
g.AddEdge(1,1)
g.AddEdge(1,2)
```

`len(s.outEdges)` would only return 1, not 2.

Reply:
You're right!